src/main/java/com/gzs/daos/inmemory/LanguageDaoInMemoryImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.gzs.daos.inmemory;
+
+import com.gzs.daos.LanguageDao;
+import com.gzs.data.GenerateTestData;
+import com.gzs.model.Language;
+
+import java.util.List;
+import java.util.Map;
+
+public class LanguageDaoInMemoryImpl extends InMemoryDao<Language> implements LanguageDao {
+    private static GenerateTestData generatedData;
+
+    static {
+        generatedData = GenerateTestData.getInstance();
+    }
+
+    public LanguageDaoInMemoryImpl() {
+        super();
+        this.dataMap = generatedData.getLanguages();
+    }
+
+    @Override
+    public List<Language> getAll() {
+        return getAllFromMap();
+    }
+
+    @Override
+    public Language get(int id) {
+        return getById(id);
+    }
+
+    @Override
+    public Language getByEnglishName(String englishName) {
+        Language language = new Language();
+        for (Map.Entry<Integer, Language> entry : dataMap.entrySet()) {
+            if (englishName.equals(entry.getValue().getEnglishName())) {
+                language = entry.getValue();
+            }
+        }
+        return language;
+    }
+
+    @Override
+    public Language getByNativeName(String nativeName) {
+        Language language = new Language();
+        for (Map.Entry<Integer, Language> entry : dataMap.entrySet()) {
+            if (nativeName.equals(entry.getValue().getNativeName())) {
+                language = entry.getValue();
+            }
+        }
+        return language;
+    }
+
+    @Override
+    public Language getByIsoCode(String iso) {
+        Language language = new Language();
+        for (Map.Entry<Integer, Language> entry : dataMap.entrySet()) {
+            if (iso.equals(entry.getValue().getIsoCode())) {
+                language = entry.getValue();
+            }
+        }
+        return language;
+    }
+
+    @Override
+    public boolean insert(Language language) {
```

Comment:
@DespotOfMorea  logic of this method is the same in each DAO. Why do they need to exist at all, as all they do is null check. I'd rename `insertT` from superclass to `insert`, made it public, moved the null check there and delete subclass overrides alltogether.